File: management-domain/Finance-department/Backend/Nodes/finance-automation-service/invoice-processing-service/src/shared/exceptions/business.exception.ts
import { HttpStatus } from '@nestjs/common';
import { BaseException } from './base.exception';

/**
 * Exception thrown when a business rule is violated
 */
export class BusinessException extends BaseException {
  constructor(message: string, errorCode: string = 'BUSINESS_ERROR', context?: Record<string, unknown>) {
    super(
      message,
      HttpStatus.UNPROCESSABLE_ENTITY,
      errorCode,
      context,
    );
  }
}

File: management-domain/Digital-marketing/Backend/Java/lead-generation-service/src/main/java/com/gogidix/digitalmarketing/shared/exception/BusinessException.java
package com.gogidix.digitalmarketing.shared.exception;

/**
 * Exception thrown when a business rule is violated.
 */
public class BusinessException extends DomainException {

    public BusinessException(String message) {
        super("BUSINESS_ERROR", message);
    }

    public BusinessException(String errorCode, String message) {
        super(errorCode, message);
    }

    public BusinessException(String message, Throwable cause) {
        super("BUSINESS_ERROR", message, cause);
    }
}
